src/modules/generator/main_impl.h

```diff
@@ -296,17 +296,12 @@ int secp256k1_pedersen_commitment_parse(const secp256k1_context* ctx, secp256k1_
 }
 
 int secp256k1_pedersen_commitment_serialize(const secp256k1_context* ctx, unsigned char *output, const secp256k1_pedersen_commitment* commit) {
-    secp256k1_ge ge;
-
     VERIFY_CHECK(ctx != NULL);
     ARG_CHECK(output != NULL);
     ARG_CHECK(commit != NULL);
 
-    secp256k1_pedersen_commitment_load(&ge, commit);
-
-    output[0] = 9 ^ secp256k1_fe_is_square_var(&ge.y);
-    secp256k1_fe_normalize_var(&ge.x);
-    secp256k1_fe_get_b32(&output[1], &ge.x);
+    output[0] = 8 ^ (commit->data[0] & 1);
+    memcpy(&output[1], &commit->data[1], 32);
```

Comment:
Couldn't you even memcpy the entire 33 bytes?

Reply:
> Couldn't you even memcpy the entire 33 bytes? 

I tried this, it does pass the tests :) 

I see `8 ^ (commit->data[0] & 1)` alternates between `1000` and `1001` binary for even/odd values of `data[0]`. 

Out of interest as cryptography noob, what is the significance of this?

Reply:
It's just a flag byte. `0x02`, `0x03` (and a few others) are already used for public keys, so whoever wrote that code took `0x08` and `0x09` to distinguish commitments from public keys.

Reply:
@real-or-random thanks!